node.go

```diff
@@ -439,15 +458,12 @@ func (n *Node) handleVoteRequest(vreq *pb.VoteRequest) bool {
 
 	deny := &pb.VoteResponse{Term: n.term, Granted: false}
 
-	// Old term, reject
-	if vreq.Term < n.term {
+	// Old term or candidate's log is behind, reject
+	if vreq.Term < n.term || !n.handler.GrantVote(vreq.LogPosition) {
 		n.rpc.SendVoteResponse(vreq.Candidate, deny)
 		return false
 	}
 
-	// Save state flag
```

Comment:
If we send a vote response, the node's state changes. Before, this only happened if the term was higher, but now it also happens if term is equal but "log" is higher. My test was failing because the node state wasn't being persisted. I guess we can still avoid saving if term is equal *and* the "log" is equal?